components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/listener/ScopeManagementAuditLogger.java

```diff
@@ -0,0 +1,128 @@
+            " : %s ";
+    private static final String SUCCESS = "Success";
+
+    private static final Log log = LogFactory.getLog(ScopeManagementAuditLogger.class);
+
+    /**
+     * This handles the OAuth scope related operations that are subscribed and publish audit logs for those operations.
+     *
+     * @param event Event.
+     * @throws IdentityEventException IdentityEventException.
+     */
+    @Override
+    public void handleEvent(Event event) throws IdentityEventException {
+
+        int tenantId = (int) event.getEventProperties().get(IdentityEventConstants.EventProperty.TENANT_ID);
+        String tenantDomain = IdentityTenantUtil.getTenantDomain(tenantId);
+        if (log.isDebugEnabled()) {
+            log.debug(event.getEventName() + " event received to ScopeManagementAuditLogger for the " +
+                    "tenant: " + tenantDomain);
+        }
+
+        if (IdentityEventConstants.Event.POST_ADD_SCOPE.equals(event.getEventName())) {
+            String scopeName =
+                    (String) event.getEventProperties().get("scopeName");
+            String initiator = getInitiatorUsername(tenantDomain);
+            String data = buildScopeData(event);
+            audit.info(String.format(AUDIT_MESSAGE, initiator, "Add-Scope", scopeName, data,
+                    SUCCESS));
+        } else if (IdentityEventConstants.Event.POST_UPDATE_SCOPE.equals(event.getEventName())) {
+            String scopeName =
+                    (String) event.getEventProperties().get("scopeName");
+            String initiator = getInitiatorUsername(tenantDomain);
+            String data = buildScopeData(event);
+            audit.info(String.format(AUDIT_MESSAGE, initiator, "Update-Scope", scopeName, data,
+                    SUCCESS));
+        } else if (IdentityEventConstants.Event.POST_DELETE_SCOPE.equals(event.getEventName())) {
+            String scopeName =
+                    (String) event.getEventProperties().get("scopeName");
+            String initiator = getInitiatorUsername(tenantDomain);
+            audit.info(String.format(AUDIT_MESSAGE, initiator, "Delete-Scope", scopeName, null,
+                    SUCCESS));
+        }
+    }
+
+    @Override
+    public String getName() {
+
+        return "ScopeManagementAuditLogger";
+    }
+
+    private String getInitiatorUsername(String tenantDomain) {
+
+        String user = CarbonContext.getThreadLocalCarbonContext().getUsername();
+        if (StringUtils.isNotBlank(user)) {
+            // Append tenant domain to username build the full qualified username of initiator.
+            user = UserCoreUtil.addTenantDomainToEntry(user, tenantDomain);
+        } else {
+            user = CarbonConstants.REGISTRY_SYSTEM_USERNAME;
+        }
+        return user;
+    }
+
+    private String buildScopeData(Event event) {
+
+        String name =
+                (String) event.getEventProperties().get("scopeName");
```

Comment:
Can't we use the `IdentityEventConstants.EventProperty.SCOPE_NAME` in here? Check other places also.